syntax = "proto3";
package trc_serialize;

message Coord {
    double lat = 1; 
    double lng = 2; 
}

message NameIndex {
	map <uint32,string> bus_name = 1;
	map <uint32, string> stop_name = 2;
} 

message Stop {
	uint32 id = 1;
	uint32 name_index = 2;
	Coord coord = 3;
}

message Bus {
	uint32 name_index = 1;
	repeated uint32 stop_id = 2;
	bool is_ring = 3;
}

message StopBuses {
	uint32 stop_id = 1;
	repeated uint32 bus_id = 2;
}

message StopDistance {
	uint32 stop_from = 1;
	uint32 stop_to = 2;
	uint32 distance = 3;
}
message Point {
	double x = 1;
	double y = 2;
}
message Color {
	uint32 r = 1;
	uint32 g = 2;
	uint32 b = 3;
	double opacity = 4;
	string name = 5;
	int32 format = 6;
}
message RenderSettings {
	uint32 width = 1;
	uint32 height = 2;
	uint32 padding = 3;
	uint32 stop_radius = 4;
	uint32 line_width = 5;
	uint32 bus_label_font_size = 6;
	Point bus_label_offset = 7;
	uint32 stop_label_font_size = 8;
	Point stop_label_offset = 9;
	Color underlayer_color = 10;
	uint32 underlayer_width = 11;
	repeated Color color_palette = 12;
}

message TransportCatalogue {
	repeated Bus bus = 1;
	repeated Stop stop = 2;
	NameIndex name_index = 3;
	repeated StopDistance stop_distance = 4;
	RenderSettings render_settings = 5;
}
